### PR TITLE
Clean up linter warnings

### PR DESCRIPTION
**Pull request by eatyourgreens**
Tidies up prop types.
Cleans up whitespace warnings, prop declarations and anonymous functions.

### CODE REVIEW

**Comment by eatyourgreens** [comment]
There's a lot of changes to read through here, but it should all just be fixing linter errors, not adding or removing functionality.

**Comment by simoneduca** [comment]
This still seems In Progress; let me know when it's ready for review.

**Comment by eatyourgreens** [comment]
Should be ready now.

File: src/components/Resource.jsx [hunk 1]
```diff
@@ -14,22 +14,17 @@ const resources = {
...
-
 function Resource(props) {
   const { contents } = props;
   const ResourceViewer = resources[props.params.resource_type];
   return (<ResourceViewer contents={contents} />);
 }
 
-Resource.propTypes = propTypes;
+Resource.propTypes = {
+  contents: PropTypes.object.isRequired,
```
Comment (simoneduca) [inline src/components/Resource.jsx]: `object` is forbidden.

File: src/containers/ProjectDashboardContainer.jsx [hunk 2]
```diff
@@ -53,38 +29,71 @@ class ProjectDashboardContainer extends Component {
...
     fetchProject: bindActionCreators(fetchProject, dispatch),
     setLanguage: bindActionCreators(setLanguage, dispatch),
     createTranslation: bindActionCreators(createTranslation, dispatch)
-  },
+  }
 });
 
-ProjectDashboardContainer.propTypes = propTypes;
+ProjectDashboardContainer.propTypes = {
+  actions: PropTypes.object.isRequired,
```
Comment (simoneduca) [inline src/containers/ProjectDashboardContainer.jsx]: `object` not allowed.

File: src/containers/ProjectListContainer.jsx [hunk 3]
```diff
@@ -22,28 +17,31 @@ class ProjectListContainer extends Component {
...
-const mapDispatchToProps = (dispatch) => ({
-  actions: bindActionCreators(projectsActions, dispatch),
+const mapDispatchToProps = dispatch => ({
+  actions: bindActionCreators(projectsActions, dispatch)
 });
 
-ProjectListContainer.propTypes = propTypes;
+ProjectListContainer.propTypes = {
+  actions: PropTypes.object.isRequired,
+  projects: PropTypes.object.isRequired
```
Comment (simoneduca) [inline src/containers/ProjectListContainer.jsx]: Same here

**Comment by eatyourgreens** [comment]
I don't think `react-fix-it` was being used, and eslint complained about it, so I've taken that out too.

File: src/containers/ProjectContentsContainer.jsx [hunk 4]
```diff
@@ -181,14 +165,37 @@ class ProjectContentsContainer extends Component {
...
 });
-const mapDispatchToProps = (dispatch) => ({
-  actions: bindActionCreators(contentsActions, dispatch),
+const mapDispatchToProps = dispatch => ({
+  actions: bindActionCreators(contentsActions, dispatch)
 });
 
-ProjectContentsContainer.propTypes = propTypes;
+ProjectContentsContainer.propTypes = {
+  actions: PropTypes.object.isRequired,
```
Comment (simoneduca) [inline src/containers/ProjectContentsContainer.jsx]: Same here

File: src/containers/ProjectContentsContainer.jsx [hunk 5]
```diff
@@ -181,14 +165,37 @@ class ProjectContentsContainer extends Component {
...
+    label: PropTypes.string,
+    value: PropTypes.string
+  }),
+  params: PropTypes.shape({
+    project_id: PropTypes.string,
+    resource_id: PropTypes.string,
+    resource_type: PropTypes.string
+  }).isRequired,
+  project: PropTypes.object.isRequired,
+  resource: PropTypes.object.isRequired
```
Comment (simoneduca) [inline src/containers/ProjectContentsContainer.jsx]: Same here

**Comment by eatyourgreens** [comment]
Re. the `object` types, I couldn't figure out how to validate those objects. If you could add pointers on that to your comments, that would be great. Where the objects are Panoptes resources, I would prefer to specify them as instances of `Resource`, but couldn't figure out how to do that either.

When components are simple helper functions for a larger component or class, like the list items, I prefer to keep them with the component that uses them. Not particularly fussed either way on this though.

File: src/containers/ProjectListContainer.jsx (hunk 3 above)
Comment (eatyourgreens) [inline src/containers/ProjectListContainer.jsx]: Any thoughts on what shape these should take?

File: src/components/LanguageSelector.jsx [hunk 6]
```diff
@@ -41,7 +42,7 @@ class LanguageSelector extends Component {
       existingLanguages.push(languageOption);
       menuLanguages = menuLanguages.filter(option => option !== languageOption);
     });
-    
+
```
Comment (eatyourgreens) [inline src/components/LanguageSelector.jsx]: This function doesn't return a value.

Comment (eatyourgreens) [inline src/components/LanguageSelector.jsx]: I guess it could perhaps return `languageOption`. Sorry, I'm not getting what the return value should be.

Comment (simoneduca) [inline src/components/LanguageSelector.jsx]: So, this is [the rule ](http://eslint.org/docs/rules/array-callback-return). Where is the function actually used? Could we extract it to its own name method?

Comment (eatyourgreens) [inline src/components/LanguageSelector.jsx]: Is 804cf59 what you're asking for here?

Comment (eatyourgreens) [inline src/components/LanguageSelector.jsx]: It's used right there, as an array map.

Comment (simoneduca) [inline src/components/LanguageSelector.jsx]: Aha, yes! Thanks, sorry for not being very eloquent.

**Comment by simoneduca** [comment]
This LGTM, I'll merge as soon as checks pass.

**Comment by eatyourgreens** [comment]
Any thoughts on how to validate Redux `actions` objects? I kind of gave up there.

**Comment by simoneduca** [comment]
I'm not sure actually. There are some libraries to do it, so perhaps it isn't trivial. Could we pass it on actions perhaps?

**Comment by eatyourgreens** [comment]
I'm happy to leave `actions` as `PropTypes.object`. I've had this problem elsewhere, when an object is a hash of arbitrary key-value pairs, so you don't know in advance what the shape will be.

**Comment by eatyourgreens** [comment]
I should have looked in [the React docs](https://facebook.github.io/react/docs/typechecking-with-proptypes.html). It might be `actions: PropTypes.objectOf(PropTypes.func)`, since every action is a function.

**Comment by eatyourgreens** [comment]
Updated in #43

**Pull request merged**